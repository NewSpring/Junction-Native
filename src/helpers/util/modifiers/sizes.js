const sizeReducer = amount =>
  properties =>
    properties.map(({ property, value }) => ({
      [property]: value * amount,
    }));

export default [
  { name: "none", reducer: sizeReducer(0) },
  { name: "xs", reducer: sizeReducer(0.25) },
  { name: "sm", reducer: sizeReducer(0.5) },
  { name: "md", reducer: sizeReducer(1) },
  { name: "lg", reducer: sizeReducer(2) },
  { name: "xl", reducer: sizeReducer(4) },
];
