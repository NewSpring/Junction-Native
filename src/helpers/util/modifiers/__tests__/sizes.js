import sizes from "../sizes";

it("should match snapshot", () => {
  expect(
    sizes.map(({ reducer, ...x }) => ({
      ...x,
      result: reducer(8),
    })),
  ).toMatchSnapshot();
});
