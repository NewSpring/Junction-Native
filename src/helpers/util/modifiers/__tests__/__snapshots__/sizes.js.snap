// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
Array [
  Object {
    "name": "none",
    "result": 0,
  },
  Object {
    "name": "xs",
    "result": 2,
  },
  Object {
    "name": "sm",
    "result": 4,
  },
  Object {
    "name": "md",
    "result": 8,
  },
  Object {
    "name": "lg",
    "result": 16,
  },
  Object {
    "name": "xl",
    "result": 32,
  },
]
`;
